src/pages/studyView/addChart/AddChart.tsx

```diff
@@ -0,0 +1,238 @@
+    store: StudyViewPageStore
+}
+
+enum CurrentOpenedDialogEnum {
+    CUSTOM_GROUPS = 'CUSTOM_GROUPS',
+    ADD_CLINICAL = 'ADD_CLINICAL',
+    ADD_GENOMIC = 'ADD_GENOMIC',
+    CLOSED = 'CLOSED'
+}
+
+type CurrentOpenedDialog = CurrentOpenedDialogEnum.CUSTOM_GROUPS | CurrentOpenedDialogEnum.CLOSED | CurrentOpenedDialogEnum.ADD_GENOMIC | CurrentOpenedDialogEnum.ADD_CLINICAL;
+
+export type ChartOption = {
+    label: string,
+    key: string,
+    disabled?: boolean,
+    selected?: boolean,
+    freq: number
+}
+
+export type AddChartFrequency = { [attrId: string]: number };
+
+@observer
+export default class AddChart extends React.Component<IAddChartProps, {}> {
+    private addCustomGroupsChartTitle = 'Add custom groups as Pie Chart';
+
+    @computed
+    get addGenomicDataChartTitle() {
+        return `Add genomic data chart (${_.filter(this.genomicDataOptions, chartOption => !chartOption.selected).length} more to add)`
+    }
+
+    @computed
+    get addClinicalDataChartTitle() {
+        return `Add clinical data chart (${_.filter(this.clinicalDataOptions, chartOption => !chartOption.selected).length} more to add)`
+    }
+
+    @observable currentOpenedDialog: CurrentOpenedDialog = CurrentOpenedDialogEnum.CLOSED;
+
+    readonly getClinicalDataCount = remoteData<AddChartFrequency>({
+        invoke: async () => {
+            const freq = this.calculateFrequency(_.reduce(this.props.store.clinicalDataWithCount.result, (acc, next, key) => {
+                acc[key] = next;
+                return acc;
+            }, {} as AddChartFrequency));
+            return freq;
+        },
+        default: {}
+    });
+
+
+    readonly getGenomicDataCount = remoteData<AddChartFrequency>({
+        invoke: async () => {
+            const freq = this.calculateFrequency(_.reduce(this.props.store.genomicDataWithCount.result, (acc, next, key) => {
+                acc[key] = next;
+                return acc;
+            }, {} as AddChartFrequency));
+            return freq;
+        },
+        default: {}
+    });
+
+    private calculateFrequency(data: AddChartFrequency): AddChartFrequency {
+        return _.reduce(data, (acc, next, key) => {
+            acc[key] = next * 100 / this.props.store.selectedSamples.result.length;
+            return acc;
+        }, {} as { [attrId: string]: number });
+    }
+
+    @bind
+    @action
+    updateCurrentOpenedDialog(dialog: CurrentOpenedDialog) {
+        this.currentOpenedDialog = dialog;
+    }
+
+    @computed
+    get genomicDataOptions(): ChartOption[] {
+        return _.reduce(_.values(this.props.store.chartMetaSet), (options, chartMeta: ChartMeta) => {
+            if (chartMeta.dataType === ChartMetaDataTypeEnum.GENOMIC) {
+                options.push({
+                    label: chartMeta.displayName,
+                    key: chartMeta.uniqueKey,
+                    disabled: false,
+                    selected: _.includes(this.selectedAttrs, chartMeta.uniqueKey),
```

Comment:
you can now use the includes method of the array prototype   this.selectedAttrs.includes(...)